herald/src/test/java/io/heraldprox/herald/sensor/datatype/RandomSourceTests.java

```diff
@@ -7,63 +7,85 @@
+import io.heraldprox.herald.sensor.datatype.random.BlockingSecureRandomSingleton;
+import io.heraldprox.herald.sensor.datatype.random.NonBlockingPRNG;
+import io.heraldprox.herald.sensor.datatype.random.RandomSource;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
 
 public class RandomSourceTests {
 
     @Test
     public void testNextInt() {
-        for (RandomSource.Method method : RandomSource.Method.values()) {
-            final RandomSource randomSource = new RandomSource(method);
+        final List<RandomSource> randomSources = Arrays.asList(
+                new NonBlockingPRNG(),
+                new BlockingSecureRandom(),
+                new BlockingSecureRandomSingleton(),
+                new BlockingSecureRandomNIST());
+        for (RandomSource randomSource : randomSources) {
+            int duplicates = 0;
             int lastValue = randomSource.nextInt();
             for (int i=0; i<1000; i++) {
                 final int value = randomSource.nextInt();
-                assertNotEquals(lastValue, value);
+                if (value == lastValue) {
+                    duplicates++;
+                }
                 lastValue = value;
             }
+            // Duplicates may occur, but should be unlikely
+            assertTrue(duplicates < 900);
         }
     }
 
     @Test
     public void testNextLong() {
-        for (RandomSource.Method method : RandomSource.Method.values()) {
-            final RandomSource randomSource = new RandomSource(method);
+        final List<RandomSource> randomSources = Arrays.asList(
+                new NonBlockingPRNG(),
+                new BlockingSecureRandom(),
+                new BlockingSecureRandomSingleton(),
+                new BlockingSecureRandomNIST());
+        for (RandomSource randomSource : randomSources) {
+            int duplicates = 0;
             long lastValue = randomSource.nextLong();
             for (int i=0; i<1000; i++) {
                 final long value = randomSource.nextLong();
-                assertNotEquals(lastValue, value);
-                lastValue = value;
-            }
-        }
-    }
-
-    @Test
-    public void testNextDouble() {
-        for (RandomSource.Method method : RandomSource.Method.values()) {
-            final RandomSource randomSource = new RandomSource(method);
-            double lastValue = randomSource.nextDouble();
-            for (int i=0; i<1000; i++) {
-                final double value = randomSource.nextDouble();
-                assertNotEquals(lastValue, value, Double.MIN_VALUE);
+                if (value == lastValue) {
+                    duplicates++;
+                }
                 lastValue = value;
             }
+            // Duplicates may occur, but should be unlikely
+            assertTrue(duplicates < 900);
```

Comment:
90% seems rather high for a tolerance.

Reply:
Updated comment to highlight this is for catching obvious errors and added uniformity test.